src/frontend/src/handler/create_sink.rs

```diff
@@ -897,6 +897,9 @@ static CONNECTORS_COMPATIBLE_FORMATS: LazyLock<HashMap<String, HashMap<Format, V
                 FileSink::<S3Sink>::SINK_NAME => hashmap!(
                     Format::Plain => vec![Encode::Parquet, Encode::Json],
                 ),
+                FileSink::<SnowflakeSink>::SINK_NAME => hashmap!(
+                    Format::Plain => vec![Encode::Parquet, Encode::Json],
```

Comment:
Prior to this PR, if `FORMAT ... ENCODE ...` is not specified by user (See the example [here](https://github.com/risingwavelabs/risingwave/blob/6d78b6e52bcca9230a29954b0d143cb51e64d4c5/integration_tests/snowflake-sink/create_sink.sql#L1)), json encode will be used while after this PR, it will be an error. Let's keep the behavior backward compatible.

Reply:
Because the encoder of the snowflake sink was previously [hardcoded as a json encoder](https://github.com/risingwavelabs/risingwave/blob/4dec2eaf45d56d1021b9b3371de770952042fe26/src/connector/src/sink/snowflake.rs#L159), json is the default here.  In other words, even without this pr, when the snowflake sink supports other encodings in the future, it will also need to add the FORMAT ... ENCODE ... field. Can we regard this pr as an enhancement to the snowflake sink, that is, changing the encoding it supports from json to json and parquet, so users need to add the FORMAT ... ENCODE ... field.


>Let's keep the behavior backward compatible

This means under this pr, we need to give a default value for FileSink's format_desc, at least the default value needs to be used when the sink is snowflake, which I think is a bit strange.

Reply:
>we need to give a default value for FileSink's format_desc when connector is snowflake

It's not hard to implement, but I'm weakly -1.

Reply:
Let's keep it paid and modify this in a separate PR if needed.